### PR TITLE
@PreferenceChange more parameter types

### LINKED ISSUE #1394: @PreferenceChange minor parameter related changes
You are supporting a Boolean parameter on the @PreferenceChange annotation (but not mentioned in documentation), and you are not supporting a

     Set<String>  

parameter (but is reported in the documentation, you probably only meant to say that Set is supported) Finally the Long parameter is not supported (and correctly not reported in the documentation).
Maybe while you are at it you can also change @SharedPref to support Double, and also make @PreferenceChange support doubles?

### PR DESCRIPTION
**Pull request by WonderCsabo**
Fixes #1394.

### CODE REVIEW

File: AndroidAnnotations/androidannotations-api/src/main/java/org/androidannotations/annotations/PreferenceChange.java [hunk 1]
```diff
@@ -42,8 +42,13 @@
  * <li>A {@link android.preference.Preference Preference} parameter to know
  * which preference was targeted by this event</li>
  * <li>An {@link Object} or {@link java.util.Set Set of strings} or
- * {@link Boolean} or {@link String} to obtain the new value of the
- * {@link android.preference.Preference Preference}</li>
+ * {@link Boolean} or {@link Float} or {@link Integer} or {@link Long} or
```
Comment (dodgex) [inline AndroidAnnotations/androidannotations-api/src/main/java/org/androidannotations/annotations/PreferenceChange.java]: not sure how it is in english, but in german we would comma seperate the words but the last. also personally i would change the order a little bit

> An Object, String, Set of Strings and also Boolean, Float, Integer, Long or thier corresponding primitive types

Comment (yDelouis) [inline AndroidAnnotations/androidannotations-api/src/main/java/org/androidannotations/annotations/PreferenceChange.java]: +1 for separating with commas and for the order.

**Comment by WonderCsabo** [comment]
Thanks for the suggestion on the JavaDoc, updated.

**Comment by WonderCsabo** [comment]
I just realized i used the new parameter validator API, but it is not available in the master branch we are merging this back. I will change this to only be on top of master and not develop.

**Comment by WonderCsabo** [comment]
Now this branch is based on master.

**Comment by WonderCsabo** [comment]
Correcting myself: of course we have to merge back to a hotfix branch, not directly to master.

File: AndroidAnnotations/androidannotations/src/main/java/org/androidannotations/handler/PreferenceChangeHandler.java [hunk 2]
```diff
@@ -79,13 +80,24 @@ protected void processParameters(HasPreferences holder, JMethod listenerMethod,
...
+
+			if (type.equals(CanonicalNameConstants.PREFERENCE)) {
 				call.arg(preferenceParam);
-			} else if (variableElement.asType().toString().equals(CanonicalNameConstants.OBJECT)) {
+			} else if (type.equals(CanonicalNameConstants.OBJECT)) {
 				call.arg(newValueParam);
+			} else if (type.equals(CanonicalNameConstants.INTEGER) || type.equals(int.class.getName()) || //
+					type.equals(CanonicalNameConstants.FLOAT) || type.equals(float.class.getName()) || //
+					type.equals(CanonicalNameConstants.LONG) || type.equals(long.class.getName())) {
+				JClass wrapperClass = type.startsWith("java") ? holder.refClass(type) : JType.parse(holder.codeModel(), type.replace(".class", "")).boxify();
```
Comment (yDelouis) [inline AndroidAnnotations/androidannotations/src/main/java/org/androidannotations/handler/PreferenceChangeHandler.java]: This is quite hard to understand. Is there another way to get the wrapper class ?

**Comment by yDelouis** [comment]
I have only one comment. Otherwise, the code is OK.

File: AndroidAnnotations/androidannotations/src/main/java/org/androidannotations/handler/PreferenceChangeHandler.java (hunk 2 above)
Comment (WonderCsabo) [inline AndroidAnnotations/androidannotations/src/main/java/org/androidannotations/handler/PreferenceChangeHandler.java]: The problem is there are two cases:
type is wrapper
type is primitive

In the first case, we can just the holder.refClass. Unfortunetaly in the second case, holder.refClass returns a low letter int class for example. JType.parse is the way to get correct primitive types, but it only expects the name, without the .class. There is no way to get "int" name from "TypeMirror", so we have to remove ".class" manually.

Comment (yDelouis) [inline AndroidAnnotations/androidannotations/src/main/java/org/androidannotations/handler/PreferenceChangeHandler.java]: OK then.

**Comment by WonderCsabo** [comment]
Rebased and merged as of 8d63943.

**Pull request closed without merging**